common/persistence/sql/workflow_state_non_maps_test.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/uber/cadence/common/persistence/sql/sqlplugin"
+)
+
+func TestUpdateSignalsRequested(t *testing.T) {
+    shardID := 1
+    domainID := serialization.MustParseUUID("bf2360c6-62c7-48a2-bd95-fe6254bca628")
+    workflowID := "test"
+    runID := serialization.MustParseUUID("bf2360c6-a2c7-48a2-bd95-fe6254bca628")
+    testCases := []struct{
+        name string
+        signalRequestedIDs []string
+        deleteSignalRequestIDs []string
+        mockSetup func(*sqlplugin.MockTx)
+        wantErr bool
+    } {
+        {
+            name: "Success case",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), []sqlplugin.SignalsRequestedSetsRow{
+                    {
+                        ShardID: int64(shardID),
+                        DomainID: domainID,
+                        WorkflowID: workflowID,
+                        RunID: runID,
+                        SignalID: "s1",
+                    },
+                    {
+                        ShardID: int64(shardID),
+                        DomainID: domainID,
+                        WorkflowID: workflowID,
+                        RunID: runID,
+                        SignalID: "s2",
+                    },
+                }).Return(nil, nil)
+                mockTx.EXPECT().DeleteFromSignalsRequestedSets(gomock.Any(), &sqlplugin.SignalsRequestedSetsFilter{
+                    ShardID: int64(shardID),
+                    DomainID: domainID,
+                    WorkflowID: workflowID,
+                    RunID: runID,
+                    SignalIDs: []string{"s3", "s4"},
+                }).Return(nil, nil)
+            },
+            wantErr: false,
+        },
+        {
+            name: "Error case - failed to insert",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                err := errors.New("some error")
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, err)
+                mockTx.EXPECT().IsNotFoundError(err).Return(true)
+            },
+            wantErr: true,
+        },
+        {
+            name: "Error case - failed to delete",
+            signalRequestedIDs: []string{"s1", "s2"},
+            deleteSignalRequestIDs: []string{"s3", "s4"},
+            mockSetup: func(mockTx *sqlplugin.MockTx) {
+                err := errors.New("some error")
+                mockTx.EXPECT().InsertIntoSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, nil)
+                mockTx.EXPECT().DeleteFromSignalsRequestedSets(gomock.Any(), gomock.Any()).Return(nil, err)
+                mockTx.EXPECT().IsNotFoundError(err).Return(true)
```

Comment:
For the NotFound error I think we can not only verify that error is returned, but also verify that error is of the expected type.

I suggest to either switch wantErr to a type check, or separate the error handling to a dedicated test that focuses on error handling.

Reply:
The check is done in a helper function which is called everywhere. I'll write a unit test for that helper function to cover the check.